### PR TITLE
Milestone: Blog app - Add forms

### PR DESCRIPTION
**Pull request by Kashif-Saleem-Ghuman**
HI Code Reviewer 👋 

🟢 In this milestone 

- [ ] Create a method called current_user in your ApplicationController to make current user data available to all controllers.
  - [ ] It will return the first user from the database.
  - [ ] Don't worry about proper authentication at this time. You will add it in the upcoming projects.
- [ ] Create forms to perform the following functions:
  - [ ] Creates a Post on behalf of the current_user (use the method that you created in your ApplicationController).
  - [ ] Create a comment on behalf of the current_user (use the method that you created in your ApplicationController).
- [ ] Allow Users to add likes to Posts.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -8,6 +8,30 @@ def show
...
+  end
+
+  def new
+    @user = current_user(params[:user_id])
+    @post = Post.new
+  end
+
+  def create
+    @user = current_user(params[:user_id])
+    @post = @user.posts.new(author: @user, title: params[:post][:title], text: params[:post][:text])
```
Comment (FrederickMih) [inline app/controllers/posts_controller.rb:21]: - [Optional] I think it is important to increase the security of data sent through forms. You did a great job by specifying which parameters are accepted and used in the post_params method but unfortunately, you didn't use the method again.

File: config/routes.rb [hunk 2]
```diff
@@ -5,7 +5,18 @@
...
+    resources :posts, only: [:index,:show, :new, :create] do
+      resources :comments, only: [:new, :create]
+    end
   end
 
+  post '/users/:user_id/posts', to: 'posts#create', as: 'create_user_post'
+  post '/users/:user_id/posts/:id/likes', to: 'likes#create', as: 'create_post_like'
+  get '/users/:user_id/posts/:post_id/comments/new', to: 'comments#new', as: 'new_post_comment'
+  post '/users/:user_id/posts/:post_id/comments', to: 'comments#create', as: 'create_post_comment'
+  get 'users/:user_id/posts/:post_id', to: 'posts#show'
```
Comment (FrederickMih) [inline config/routes.rb:17]: - [Optional] I think most of these routes are found in the resourceful routes above. It is an important practice to keep our codebase DRY.

**Review by FrederickMih — APPROVED** [review]
## Hi :raising_hand:  @Kashif-Saleem-Ghuman , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights :green_circle: 


- Nice forms for creating posts and comments :+1: 
- Fantastic commit history :+1: 
- Descriptive README.md file :+1: 

## Optional suggestions

- In the users' table, the column bio was named **Bio** while the other columns are in lowercase. I think this can be an issue for someone new to your codebase. Let's always make sure our naming conventions are homogenous



_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Kashif-Saleem-Ghuman** [comment]
@FrederickMih  Thankyou for approving and providing valuable feedback 👍

**Pull request merged**